llama_index/evaluation/pairwise.py

```diff
@@ -0,0 +1,116 @@
+"""Pairwise evaluation."""
+
+from llama_index.evaluation.base import BaseEvaluator, EvaluationResult
+from llama_index import ServiceContext
+from typing import Optional, Union, Sequence, Any
+from llama_index.prompts import (
+    BasePromptTemplate,
+    ChatMessage,
+    ChatPromptTemplate,
+    MessageRole,
+    PromptTemplate,
+)
+
+
+DEFAULT_SYSTEM_TEMPLATE = """
+You are an expert evaluation system for a question answering chatbot.
+
+You are given the following information:
+- a user query, 
+- a reference answer, and
+- a candidate answer.
+
+Your job is to output whether the candidate answer is better than the reference \
+answer in answering the user query. \
+'YES' means the candidate answer is better, 'NO' means the \
+reference answer is better, and 'TIE' means they are equally good.
+
+Please output two lines: 
+- first line: the word YES, NO, or TIE
+- second line: a short explanation for your decision.
+"""
+
+DEFAULT_USER_TEMPLATE = """
+## User Query
+{query}
+
+## Reference Answer
+{reference_answer}
+
+## Candidate Answer
+{generated_answer}
+"""
+
+DEFAULT_EVAL_TEMPLATE = ChatPromptTemplate(
+    message_templates=[
+        ChatMessage(role=MessageRole.SYSTEM, content=DEFAULT_SYSTEM_TEMPLATE),
+        ChatMessage(role=MessageRole.USER, content=DEFAULT_USER_TEMPLATE),
+    ]
+)
+
+
+class PairwiseComparisonEvaluator(BaseEvaluator):
```

Comment:
Should this name be more like `LLMPairwiseComparisonEvaluator` ? I can imagine a more efficient pairwise evaluator using cross-encoders or something else 🤔

Reply:
yeah makes sense, can rename

Reply:
actually thought about it, might keep for now. reason is our other eval modules also aren't prefixed by LLM (the correctness evaluator for instance is basically the LLM version of the semantic similarity evaluator)